src/main/kotlin/com/cognifide/gradle/aem/instance/tail/Log.kt

```diff
@@ -47,33 +55,42 @@ class Log(
 
         private const val LOG_PATTERN = "$TIMESTAMP\\s$LEVEL\\s$SOURCE\\s$MESSAGE"
 
-        private const val DATE_TIME_FORMAT = "dd.MM.yyyy HH:mm:ss.SSS"
+        private const val LOGS_SEPARATOR = "  "
+
+        private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm:ss.SSS")
 
-        private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(DATE_TIME_FORMAT)
+        private val PRINT_DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS")
 
-        fun create(logLines: List<String>): Log {
+        fun create(instance: InstanceLogInfo, logLines: List<String>): Log {
             if (logLines.isEmpty() || logLines.first().isBlank()) {
                 throw InstanceTailerException("Passed log entry is empty!")
             }
 
             val fullLog = logLines.joinToString("\n")
-            val result = matchLogLine(logLines.first())
 
-            when (result) {
+            when (val result = matchLogLine(logLines.first())) {
                 null -> throw InstanceTailerException("Passed text is not a log entry\nPattern:\n$LOG_PATTERN\nText:\n${logLines.first()}")
                 else -> {
                     val (timestamp, level, source, message) = result.destructured
                     val followingMessageLines = logLines.slice(1 until logLines.size)
-                    return Log(fullLog, parseTimestamp(timestamp), level, source, listOf(message) + followingMessageLines)
+                    return Log(
+                            instance,
+                            fullLog,
+                            parseTimestamp(timestamp, instance),
+                            level,
+                            source,
+                            listOf(message) + followingMessageLines
+                    )
                 }
             }
         }
 
         fun isFirstLineOfLog(text: String) = matchLogLine(text) != null
 
-        fun parseTimestamp(timestamp: String): LocalDateTime {
-            return LocalDateTime.parse(timestamp, DATE_TIME_FORMATTER)
-                    ?: throw InstanceTailerException("Invalid timestamp in log:\n$timestamp\n required format: $DATE_TIME_FORMAT")
+        fun parseTimestamp(timestamp: String, instance: InstanceLogInfo = InstanceLogInfo.none()): ZonedDateTime {
+            return LocalDateTime.parse(timestamp, DATE_TIME_FORMATTER).atZone(instance.zoneId)
+                    ?: throw InstanceTailerException("Invalid timestamp in log:\n$timestamp" +
```

Comment:
you have context of instance here, so you could include its name in exception log to more easily find  it on error...